### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by AlyAtalla**
Added a new JavaScript file and import it as a module:
it contains methods related to the status updates (completed: true / false).
Added event listener to the checkbox (change).
Updated items object's value for completed key upon user actions.
Implemented a function for the "Clear all completed" button using filter() method).
Stored the updated array of items in local storage, so the user gets the correct list of values after the page reloads.

### CODE REVIEW

File: src/modules/interactive.js
Comment (AbelG101) [inline src/modules/interactive.js:1]: - [x] There seems to be a bug in the implementation of the "Completed" status as shown in the video below whenever the user clicks the checkbox for one task the entire task is removed from the view and another task has to be added to restore them I also noticed only tasks at the bottom are properly getting ticked off, whenever I remove the tasks at the top the entire tasks are permanently removed, please make sure to fix this.
- Here's how it behaves:
  firefox_iClIRN6pM9

Comment (AbelG101) [inline src/modules/interactive.js:1]: - [x] The "Clear All tasks" button is not functioning as expected as well as it removes unchecked tasks and keeps some of them, the expected behavior is to clear only the tasks that have the `completed status` set to true so please make sure to resolve this bug.
- Here's how it behaves:
  firefox_e2LAVsgIkZ

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @AlyAtalla 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- UI is looking great ✔
- Correct use of GitHub Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
   <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.1/css/all.min.css"
     integrity="sha512-KfkfwYDsLkIlwQp6LFnl8zNdLGxu9YAA1QvwINks4PhcElQSvqcyVLLD9aMhXd13uQjoXtEKNosOWaZqXgel0g=="
     crossorigin="anonymous" referrerpolicy="no-referrer" />
-  <link rel="stylesheet" href="styles.css">
+  <link rel="stylesheet" href="style.css">
 </head>
 
 <body>
```
Comment (abdulhamiid) [inline src/index.html:15]: Your design
image

Template
image

[OPTIONAL] - To improve the UI design, please modify your CSS to style the list according to the [captured video](https://www.youtube.com/watch?v=AcUd-_Yjjqg).

- The icons' position as highlighted above
- I noticed you used a `check-box` 👀for the icon in the submit button which does not match the template, please correct that. 🤗

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @AlyAtalla 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Use webpack to bundle JavaScript.
- ✔️ Use ES6 modules to write modular JavaScript.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**